pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -23,8 +23,13 @@ import (
 
 // PipelineSpec defines the desired state of PipeLine.
 type PipelineSpec struct {
-	Tasks      []PipelineTask `json:"tasks"`
-	Generation int64          `json:"generation,omitempty"`
+	Tasks []PipelineTask `json:"tasks"`
+	// Time after which the build times out. Defaults to never.
+	// Specified build timeout should be less than 24h.
```

Comment:
is there a specific reason why less than 24h? (e.g. something we can link to) otherwise i wonder if we really should set an upper bound at all :thinking: people might want to do some crazy things

(ps. maybe update the comment here to say "pipeline" instead of "build"?)

Reply:
Good question - I just assumed we'd want a max, but I'll yank it out for now.